CHANGELOG.md

```diff
@@ -3,6 +3,150 @@
 All notable changes to this project will be documented in this file.
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 7.0.0-alpha.0
+
+_Nov 9, 2023_
+
+We'd like to offer a big thanks to the 11 contributors who made this release possible. Here are some highlights ✨:
+
+- ✨ v7 Alpha release
+- 🌍 Improve Czech (cs-CZ) locale
+
+From https://github.com/mui/mui-x/pull/10875
+
+#### Breaking changes
+
+Types for `slots` and `slotProps` got renamed by removing the "Component" which is meaningless for charts.
+Except if you imported those types, to create a wrapper, you should not be impacted by this breaking change.
+
+Here is an example of the renaming for the `<ChartsLegend />` component.
+
+```diff
+-ChartsTooltipSlotsComponent
++ChartsTooltipSlots
+
+-ChartsTooltipSlotComponentProps
++ChartsTooltipSlotProps
+```
+
+### Data Grid
+
+#### `@mui/x-data-grid@7.0.0-alpha.0`
+
+- [DataGrid] Fix for error thrown when removing skeleton rows, after sorting is applied (#10807) @benjaminbialy
+- [DataGrid] Fix: `undefined` slot value (#10937) @romgrk
+- [DataGrid] Print selected rows by default (#10846) @cherniavskii
+- [DataGrid] Remove deprecated `components` and `componentsProps` (#10911) @MBilalShafi
+- [DataGrid] Remove legacy filtering API (#10897) @cherniavskii
+- [DataGrid] Fix keyboard navigation for actions cell with disabled buttons (#10882) @michelengelen
+- [DataGrid] Added a recipe for using non-native select in filter panel (#10916) @michelengelen
+- [DataGrid] Added a recipe to style cells without impacting the aggregation cells (#10913) @michelengelen
+- [l10n] Improve Czech (cs-CZ) locale (#10949) @luborepka
```

Comment:
Nitpick: Weren't we porting back the l10n changes during the v5 -> v6 transition? 🤔

Reply:
Yep ... I would expect that as well ... maybe that got lost somewhere?

Reply:
Yes we were

Reply:
Maybe the current strategy is flawed.
Wouldn't it be better if we target `master` for bugfixes, translations, etc and before each release we merge `master` into `next`? Seems more feasible and it's harder to "lose" PRs

Reply:
Your suggestion does have a lot of merit. 🤔 
What do others think @mui/x?

Reply:
I think it will cause the PR not appearing in the next changelog like happened with https://github.com/mui/mui-x/pull/10543
The strategy of merging everything in next and `cherry-picking` the `l10n`, `bug fixes` and other things we decide to port seem the most suitable one, as there will be a single source of truth (next)
Regarding this specific change, we can push it on to the next master/v6 release, I think it won't hurt to release it one week later.

Reply:
Nice suggestion, I am not against it, but I am still inclining towards providing more visibility of what's delivered in current release directly in the specific release page.

Since our changelogs comprises of changes on multiple packages, it might get cumbersome for the users to know what specific changes are coming from the master release for their specific package, and they might miss the ones coming from the master.

Another reason this might cause some issues is, there were cases in past when we delivered some changes on master which we did not deliver on next, (for example an API which we deprecated in the new major but needs some fixing in current major) so saying `Every change from [v6.18.1](#6.18.1) and the below.` would not possibly be true in that case. 🤷‍♂️

Reply:
Nice catch!
Well ... we can still go with the cherry-pick-to-master approach. We just need to be more careful to include all changes we want on master as well.

Not so sure about #10916 and #10913 as well, since those are still viable for `v6` as well.

Reply:
If I remember correctly from what we discussed with the team the other day and the strategy we followed last year, from now on, we should only target important bug fixes (minor ones could possibly be skipped), localization changes and those things which are necessary for v6. Other low priority things like these recipes, new features, improvements and of coarse the breaking changes should be limited to the new major, basically to create some incentive for migration.
So, I'd vote not to port these recipes to v6.

Reply:
> I think it will cause the PR not appearing in the next changelog like happened with #10543

Correct me if I'm wrong, but I think that the problem why individual changes are not picked up is the fact that we default to do a squash commit and merge.
If we decide to go with the proposal, then this particular PR would need to be merged regularly.
Doing a regular merge will include all the relevant merge commits in the branch and thus the generated changelog should be complete. 🤔 
WDYT?

Reply:
That should do the trick! 
And since this is only done occasionally it is only a slight deviation from the normal process!

I am pretty confident that this could be the better approach, but we'll see what the others have to say about it

Reply:
> Another reason this might cause some issues is, there were cases in past (and could be in the future) when we delivered some changes on master which we did not deliver on next, (for example a deprecated API which we removed in the new major but needs some fixing in current major) so saying Every change from [v6.18.1](#6.18.1) and the below. would not possibly be true in that case.

Even if we manage to pick up the changes individually (it will include the release PR too I guess 😄), wouldn't ⬆️ still be a problem? 🤔

Reply:
I think we cannot circumvent manual filtering and adjustments in any case 🤷🏼

Reply:
So, to summarize.
It seems that we are leaning toward sticking with the current approach of merging everything to `next` and cherry-picking manually only what's necessary.
If the problem arises once more - let's re-evaluate.
Because, as we've discussed, both approaches have their cons. Having the flexibility of choosing what we actually need sounds nice as well. 👌 
Correct me, if I made the wrong assumption? ℹ️

Reply:
I think yes, that's what we'll be adapting to as of now, right @michelengelen?

Reply:
I guess that's the most reasonable approach for now! 👍🏻